docs/groovy/conceptual/positional-index-safety.md

```diff
@@ -0,0 +1,334 @@
+---
+title: Positional indices
+---
+
+When working with Deephaven tables, you may encounter positional indices (`i`, `ii`) and row keys (`k`) in formulas. Understanding when these are safe to use is crucial for avoiding incorrect results and runtime errors.
+
+## The fundamental rule
+
+**Positional indices (`i`, `ii`, `k`) and column array variables are unsafe on refreshing tables** and will cause `IllegalArgumentException` errors by default. The engine blocks these formulas to prevent incorrect results.
```

Comment:
This is inaccurate. You can do:

```python
tt = time_table("PT1s").update("X = ii")
```

and no error is raised, despite it being a refreshing table.